x-pack/plugins/lens/public/app_plugin/tags_saved_object_save_modal_dashboard_wrapper.tsx

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, useState, useMemo, useCallback } from 'react';
+import { OnSaveProps } from '../../../../../src/plugins/saved_objects/public';
+import {
+  DashboardSaveModalProps,
+  SavedObjectSaveModalDashboard,
+} from '../../../../../src/plugins/presentation_util/public';
+import { SavedObjectTaggingPluginStart } from '../../../saved_objects_tagging/public';
+
+type TagEnhancedSavedObjectSaveModalDashboardProps = Omit<DashboardSaveModalProps, 'onSave'> & {
+  initialTags: string[];
+  savedObjectsTagging?: SavedObjectTaggingPluginStart;
+  onSave: (props: OnSaveProps & { dashboardId?: string | null; newTags?: string[] }) => void;
+};
+
+export const TagEnhancedSavedObjectSaveModalDashboard: FC<TagEnhancedSavedObjectSaveModalDashboardProps> = ({
```

Comment:
This file shares a lot of code with `x-pack/plugins/lens/public/app_plugin/tags_saved_object_save_modal_origin_wrapper.tsx` - can we somehow unify them?

Reply:
I thought about that, for sure. My biggest concern is I kinda want to make it easy to back out of changes if we have to end up deciding to make big adjustments to the time-to-visualize flow after we release. I can def go either way here, though.

Reply:
I see, then let’s move the “renderModal” logic into a separate component which calls the respective modal in the right scenarios. This keeps that stuff out of the app component (it already way too big anyway) and doesn’t over-abstract stuff which will likely change in the future.